### PR TITLE
[BUGFIX beta] Ensure `parentView` includes yielding component.

### LINKED ISSUE #11170: [Glimmer] Nested components no longer have access to their parent?
I recently tried upgrading [ivy-tabs](https://github.com/ivyapp/ivy-tabs) to Ember 1.13 and ran into some issues around changes to `parentView` in components. Prior to 1.13, a nested component could access its outer component through `parentView`. I was using this in ivy-tabs to register an `ivy-tab` component with the `ivy-tab-list` that contained it, among other things:

```handlebars
{{#ivy-tab-list}}
  {{#ivy-tab}}Tab 1{{/ivy-tab}}
  {{#ivy-tab}}Tab 2{{/ivy-tab}}
  ...
{{/ivy-tab-list}}
```

It seems that with the 1.13 upgrade, `parentView` is now always the "routable" view that rendered the component, and I don't see a way for components to access each other at all. Is my example an intended use case? If not, is there some other way to do what I want?

### PR DESCRIPTION
**Pull request by rwjblue**
Fixes #11170.

### CODE REVIEW

**Comment by truenorth** [comment]
Thanks for this! It unblocks a large amount of glimmer testing

File: packages/ember-htmlbars/tests/integration/component_invocation_test.js [hunk 1]
```diff
@@ -575,6 +575,44 @@ QUnit.test("comopnent should rerender when a property is changed during children
 
 });
 
+QUnit.test("components in template of a yielding component should have the proper parentView", function() {
+  var outer, innerTemplate, innerLayout;
+
+  registry.register('component:x-outer', Component.extend({
+    init() {
+      this._super(...arguments);
```
Comment (mixonic) [inline packages/ember-htmlbars/tests/integration/component_invocation_test.js]: Fun fact: This transpiles directly to:

```
this._super.apply(this, arguments);
```

Which is **awesome**. Elsewhere in ember we use (and I am far less crazy about):

```
registry.register('component:x-outer', Component.extend({
   init(...args) {
    this._super(...args);
    outer = this;
      }
 }));
```

Which builds an intermediate array of args before unrolling.

**Comment by truenorth** [comment]
when can we see this in canary/beta channel?

**Comment by rwjblue** [comment]
Should already be available from bower for both beta and canary channels.

**Comment by tpitale** [comment]
Did this get reverted before 2.0.0-beta.1? I can't seem to find this change, and I think I'm seeing this causing an error in https://github.com/thefrontside/emberx-select/issues/44 in this code https://github.com/thefrontside/emberx-select/blob/54f9aee9e9b53c586623646a7f8966ce26321736/addon/components/x-option.js#L58

**Pull request merged**